src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -51,7 +51,16 @@ public function __construct($classLoader)
         }
 
         if (!isset(self::$caseCheck)) {
-            self::$caseCheck = false !== stripos(PHP_OS, 'win') ? (false !== stripos(PHP_OS, 'darwin') ? 2 : 1) : 0;
+            if(!file_exists(strtolower(__FILE__))) {
+                // filesystem is case sensitive
+                self::$caseCheck = 0;
+            } elseif(realpath(strtolower(__FILE__)) === __FILE__) {
```

Comment:
Am I wrong, or the old code never could return "2" ? PHP_OS contains "darwin" or "win", not both.